Source/Details/ASDataController.mm

```diff
@@ -496,6 +506,15 @@ - (void)updateWithChangeSet:(_ASHierarchyChangeSet *)changeSet
         _visibleMap = newMap;
 
         [_delegate dataController:self didUpdateWithChangeSet:changeSet];
+        
+        // Step 5: If the environment changed mid-update, notify all visible elements
+        __weak id<ASTraitEnvironment> newEnvironment = [self.environmentDelegate dataControllerEnvironment];
+        ASPrimitiveTraitCollection newPrimitiveTraitCollection = [newEnvironment primitiveTraitCollection];
+        if (! ASPrimitiveTraitCollectionIsEqualToASPrimitiveTraitCollection(primitiveTraitCollection, newPrimitiveTraitCollection)) {
```

Comment:
Let's just name these local variables `oldTraitCollection` and `newTraitCollection`.